event_session/models/event.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 David Vidal<david.vidal@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class EventEvent(models.Model):
+    _inherit = 'event.event'
+
+    session_ids = fields.One2many(
+        comodel_name='event.session',
+        inverse_name='event_id',
+        string='Sessions',
+    )
+    sessions_count = fields.Integer(
+        compute='_compute_sessions_count',
+        string='Total event sessions',
+        store=True,
+    )
+
+    @api.multi
+    @api.depends('session_ids')
+    def _compute_sessions_count(self):
+        for event in self:
+            event.sessions_count = len(event.session_ids)
+
+    @api.multi
+    @api.constrains('seats_max', 'seats_available')
+    def _check_seats_limit(self):
+        if self.sessions_count < 1:
```

Comment:
Use `if not event.session_ids:` instead (being `event` the variable for each record.